### PR TITLE
Transformations: Add Delimiter format option to Extract fields

### LINKED ISSUE #97337: Transformations: Extract to fields support comma seperated values
**Why is this needed**:

Multiple choice questions in Google Forms show up as a list of comma separated values, which is a good use case.

**What would you like to be added**:

Either improve the Regex matches, or make on of the auto options support CSV. The actual extracted value would probably be a boolean.

**Who is this feature for?**

Editors

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

Extracts comma separated values as an option to extract to fields

**Why do we need this feature?**

Google Forms outputs multiple choice questions as comma separated values

**Who is this feature for?**

Editors

**Which issue(s) does this PR fix?**:

Closes #97337

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/transformers/extractFields/fieldExtractors.ts [hunk 1]
```diff
@@ -146,7 +165,7 @@ const extAuto: FieldExtractor = {
       for (const parse of parsers) {
         try {
           const r = parse(v);
-          if (r != null) {
+          if (r != null || r !== undefined) {
```
Comment (leeoniya) [inline public/app/features/transformers/extractFields/fieldExtractors.ts:168]: ?

File: public/app/features/transformers/extractFields/fieldExtractors.ts [hunk 2]
```diff
@@ -133,7 +133,26 @@ const extLabels: FieldExtractor = {
...
+      if (parts.length < 1) {
+        return undefined;
+      }
+      return parts.reduce((acc: Record<string, boolean>, part: string) => {
+        acc[part] = true;
+        return acc;
+      }, {});
+    };
+  },
+};
```
Comment (leeoniya) [inline public/app/features/transformers/extractFields/fieldExtractors.ts:153]: maybe something at avoids function loops and extra array allocations:

```suggestion
  getParser: ({ delimiter = ',' }) => {
    const splitRegExp = new RegExp(`\\s*${escapeStringForRegex(delimiter)}\\s*`, 'g');

    return (raw: string) => {
      const parts = raw.trim().split(splitRegExp);

      if (parts.length === 0) {
        return;
      }

      const acc: Record<string, boolean> = {};
      for (let part of parts) {
        acc[part] = true;
      }
      return acc;
    };
  },
```

File: public/app/features/transformers/extractFields/fieldExtractors.ts (hunk 1 above)
Comment (tskarhed) [reply public/app/features/transformers/extractFields/fieldExtractors.ts:168]: Oops, left this when troubleshooting the auto extractor

Comment (tskarhed) [reply public/app/features/transformers/extractFields/fieldExtractors.ts:168]: But why does the parser check for null, when it is only undefined that is allowed to be returned? I assume the definition of success here is not throwing an error?

**Review by leeoniya — APPROVED** [review]
thanks!

File: public/app/features/transformers/extractFields/fieldExtractors.ts [hunk 3]
```diff
@@ -133,7 +133,26 @@ const extLabels: FieldExtractor = {
...
+  description: 'Splits at delimited values, such as commas',
+  getParser: ({ delimiter = ',' }) => {
+    const splitRegExp = new RegExp(`\\s*${escapeStringForRegex(delimiter)}\\s*`, 'g');
+
+    return (raw: string) => {
+      // Try to split delimited values
+      const parts = raw.split(splitRegExp);
+      const acc: Record<string, number> = {};
+      for (const part of parts) {
+        acc[part.trim()] = 1;
```
Comment (leeoniya) [inline public/app/features/transformers/extractFields/fieldExtractors.ts:148]: you shouldn't need `trim()` here, right?

File: public/app/features/transformers/extractFields/fieldExtractors.ts (hunk 1 above)
Comment (tskarhed) [reply public/app/features/transformers/extractFields/fieldExtractors.ts]: I knew about the nullish coalescing operator, but I had no idea you could do a nullish check by using a non-strict statement. Thanks!

File: public/app/features/transformers/extractFields/ExtractFieldsTransformerEditor.tsx [hunk 4]
```diff
@@ -115,6 +122,19 @@ export const extractFieldsTransformerEditor = ({
       {options.format === FieldExtractorID.JSON && (
         <JSONPathEditor options={options.jsonPaths ?? []} onChange={onJSONPathsChange} />
       )}
+      {options.format === FieldExtractorID.Delimiter && (
+        <InlineFieldRow>
+          <InlineField label="Delimiter" labelWidth={16}>
+            <Combobox
```
Comment (leeoniya) [inline public/app/features/transformers/extractFields/ExtractFieldsTransformerEditor.tsx:128]: nice :)

**Pull request merged**